src/pymedphys/_labs/paulking/profile.py

```diff
@@ -24,27 +24,56 @@
 # program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
 
 
+from copy import deepcopy
 from typing import Callable
 
 import numpy as np
 from scipy import interpolate
 import matplotlib.pyplot as plt
 
+import xarray as xr
+
 from ...libutils import get_imports
-from ...xarray import XArrayComposition
+# from .._level1.coreobjects import _PyMedPhysBase
 IMPORTS = get_imports(globals())
 
 
 NumpyFunction = Callable[[np.ndarray], np.ndarray]
 
-
+#
 # pylint: disable = C0103, C0121
 
 
-class Dose1D(XArrayComposition):
+class DoseBase():
```

Comment:
This class here I had pulled out into XArrayComposition, it seems in your copy pasting you may have pulled from a version before that merge? This is in your labs directory so you can do what you want, just want to make sure this is what you want...

Reply:
Is XArrayComposition something that you wrote, or is it part of xarray?

Reply:
XArrayComposition is just something I wrote. It's within `pymedphys.xarray`